engine/src/main/java/org/terasology/engine/core/modes/loadProcesses/CreateWorldEntity.java

```diff
@@ -55,26 +61,27 @@ public String getMessage() {
 
     @Override
     public boolean step() {
-        if (worldEntityExists()) {
-            EntityRef worldEntity = getExistingWorldEntity();
-            configureChunkProvider(worldEntity);
+        this.entityManager = context.get(EntityManager.class);
+        this.worldGenerator = context.get(WorldGenerator.class);
+        this.config = context.get(Config.class);
+        this.chunkProvider = context.get(ChunkProvider.class);
+        this.worldConfigurator = worldGenerator.getConfigurator();
+
+        Iterator<EntityRef> worldEntityIterator = entityManager.getEntitiesWith(WorldComponent.class).iterator();
+        if (worldEntityIterator.hasNext()) {
+            EntityRef worldEntity = worldEntityIterator.next();
+            worldEntityIterator.forEachRemaining(w -> logger.warn("Ignored extra world {}", w));
+            chunkProvider.setWorldEntity(worldEntity);
             useConfigurationOfCurrentWorld(worldEntity);
         } else {
             EntityRef worldEntity = createWorldPoolsAndEntity();
-            configureChunkProvider(worldEntity);
-            WorldConfigurator worldConfigurator = useConfigurationFromConfig();
-            configurateWorldEntity(worldConfigurator, worldEntity);
+            chunkProvider.setWorldEntity(worldEntity);
+            useConfigurationFromConfig(worldEntity);
         }
 
         return true;
     }
 
-    private boolean worldEntityExists() {
-        EntityManager entityManager = context.get(EntityManager.class);
-        Iterable<EntityRef> worldEntityIterator = entityManager.getEntitiesWith(WorldComponent.class);
-        return worldEntityIterator.iterator().hasNext();
```

Comment:
I think I removed it because it added more logic to something that seemed straightforward to me. Its reused logic for the sake of decomposing a class.  I would probably opt more for adding comments why hasNext is necessary. i.e `if there is no world entity then we need to create one` then trying to derive it from more descriptive method names.  that also personal preference.